File: random_and_timer_cs/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace random_and_timer_cs
{
    public partial class Form1 : Form
    {
        private readonly Random generator = new Random();
        private readonly Timer timer = new Timer();
        private readonly int interval = 1500;
        private readonly int precision = 1;
        private static readonly int defaultN = 4;
        private static int n_data = defaultN;
        private int[] current_data;
        private float current_mean = 0.0f;

        public Form1()
        {
            InitializeComponent();
        }
        public static int getDefaultN()
        {
            return defaultN;
        }
        private void InitDataAndMean()
        {
            this.current_data = new int[n_data];
            this.current_mean = 0.0f;
        }
        private void GenerateDataAndComputeMean()
        {
            this.InitDataAndMean();
            for (int i = 0; i < n_data; i++)
            {
                var current_data = this.generator.Next(0, 100);
                this.current_data[i] = current_data;
                this.current_mean = (this.current_mean * i + current_data) / (i + 1);
            }
        }

        private void ShowDataAndMean()
        {
            for (int i = 0; i < n_data; i++)
            {
                this.richTextBox1.Text += "(" + (i + 1) + ")" + " " + this.current_data[i].ToString("n" + this.precision) + "\n";
            }
            this.richTextBox1.Text += "mean: " + this.current_mean.ToString("n" + this.precision) + "\n\n";
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.timer1.Interval = interval;
            this.timer1.Start();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.richTextBox1.Clear();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            this.GenerateDataAndComputeMean();
            this.ShowDataAndMean();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            this.timer1.Stop();
        }

        private void numericUpDown1_ValueChanged(object sender, EventArgs e)
        {
            if (this.numericUpDown1.Value <= 0)
            {
                this.numericUpDown1.Value = defaultN;
            }
            else
            {
                n_data = (int)this.numericUpDown1.Value;
            }
        }

        private void button4_Click(object sender, EventArgs e)
        {
            this.timer1.Stop();
            this.GenerateDataAndComputeMean();
            this.ShowDataAndMean();
        }

    }
}

File: random_and_timer_vb/Form1.vb
Public Class Form1
    Private ReadOnly generator As Random = New Random()
    Private ReadOnly interval As Integer = 1500
    Private ReadOnly precision As Integer = 1
    Private Shared ReadOnly defaultN As Integer = 4
    Private n_data As Integer = defaultN
    Private current_data As Integer()
    Private current_mean As Single = 0.0F

    Public Shared Function getDefaultN() As Integer
        Return defaultN
    End Function

    Private Sub InitDataAndMean()
        Me.current_data = New Integer(n_data - 1) {}
        Me.current_mean = 0.0F
    End Sub

    Private Sub GenerateDataAndComputeMean()
        Me.InitDataAndMean()

        For i As Integer = 0 To Me.n_data - 1
            Dim current_data = Me.generator.[Next](0, 100)
            Me.current_data(i) = current_data
            Me.current_mean = (Me.current_mean * i + current_data) / (i + 1)
        Next
    End Sub

    Private Sub ShowDataAndMean()
        For i As Integer = 0 To n_data - 1
            Me.RichTextBox1.Text += "(" & (i + 1) & ")" & " " & Me.current_data(i).ToString("n" & Me.precision) & vbLf
        Next

        Me.RichTextBox1.Text += "mean: " & Me.current_mean.ToString("n" & Me.precision) & vbLf & vbLf
    End Sub

    Private Sub Button1_Click_1(sender As Object, e As EventArgs) Handles Button1.Click
        Me.Timer1.Interval = interval
        Me.Timer1.Start()
    End Sub

    Private Sub Button2_Click_1(sender As Object, e As EventArgs) Handles Button2.Click
        Me.RichTextBox1.Clear()
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs)
    End Sub

    Private Sub Timer1_Tick_1(sender As Object, e As EventArgs) Handles Timer1.Tick
        Me.GenerateDataAndComputeMean()
        Me.ShowDataAndMean()
    End Sub

    Private Sub Button3_Click_1(sender As Object, e As EventArgs) Handles Button3.Click
        Me.Timer1.[Stop]()
    End Sub

    Private Sub NumericUpDown1_ValueChanged_1(sender As Object, e As EventArgs) Handles NumericUpDown1.ValueChanged
        If Me.NumericUpDown1.Value <= 0 Then
            Me.NumericUpDown1.Value = defaultN
        Else
            Me.n_data = CInt(Me.NumericUpDown1.Value)
        End If
    End Sub

    Private Sub Button4_Click_1(sender As Object, e As EventArgs) Handles Button4.Click
        Me.Timer1.[Stop]()
        Me.GenerateDataAndComputeMean()
        Me.ShowDataAndMean()
    End Sub
End Class
